### PR TITLE
Cleanups

### PR DESCRIPTION
**Pull request by swansontec**
### CHANGELOG

 none

### Dependencies

 none

### Description

Remove a bunch of dead code and improve type-safety throughout this repo.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204464718377066

### CODE REVIEW

File: src/react-native-io.js [hunk 1]
```diff
@@ -1,7 +1,11 @@
 import bridge from 'react-native-mymonero-core'
 
+// We are stealing and undocumented internal property off of the SDK.
```
Comment (swansontec) [inline src/react-native-io.js]: "We are stealing AN", not "and".

File: src/xmrEngine.js [hunk 2]
```diff
@@ -90,7 +90,7 @@ export class MoneroEngine {
     this.walletLocalDataDirty = false
     this.transactionsChangedArray = []
     this.walletInfo = walletInfo
-    this.walletId = walletInfo.id ? `${walletInfo.id} - ` : ''
```
Comment (samholmes) [inline src/xmrEngine.js:95]: Has `walletId` always been suffixed with a `-` before? How has this not created bugs? Isn't this data written to disk?

**Review by samholmes — APPROVED** [review]

File: src/xmrEngine.js (hunk 2 above)
Comment (swansontec) [reply src/xmrEngine.js:88]: Yes, this has been a long-standing bug. However, the edge-core-js shim code has been saving us this whole time, by stomping this with the correct ID on the way out the door.

**Pull request merged**